prometheus/src/main/java/org/opensearch/sql/prometheus/functions/scan/QueryExemplarsFunctionTableScanOperator.java

```diff
@@ -0,0 +1,75 @@
+package org.opensearch.sql.prometheus.functions.scan;
+
+import java.io.IOException;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.Locale;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.json.JSONArray;
+import org.opensearch.sql.data.model.ExprValue;
+import org.opensearch.sql.executor.ExecutionEngine;
+import org.opensearch.sql.prometheus.client.PrometheusClient;
+import org.opensearch.sql.prometheus.functions.response.QueryExemplarsFunctionResponseHandle;
+import org.opensearch.sql.prometheus.request.PrometheusQueryExemplarsRequest;
+import org.opensearch.sql.storage.TableScanOperator;
+
+@RequiredArgsConstructor
+public class QueryExemplarsFunctionTableScanOperator extends TableScanOperator {
+
+  private final PrometheusClient prometheusClient;
+
+  @Getter
+  private final PrometheusQueryExemplarsRequest request;
+  private QueryExemplarsFunctionResponseHandle queryExemplarsFunctionResponseHandle;
+  private static final Logger LOG = LogManager.getLogger();
+
+  @Override
+  public void open() {
+    super.open();
+    this.queryExemplarsFunctionResponseHandle
+        = AccessController
+          .doPrivileged((PrivilegedAction<QueryExemplarsFunctionResponseHandle>) () -> {
+            try {
+              JSONArray responseArray = prometheusClient.queryExemplars(
+                  request.getQuery(),
+                  request.getStartTime(), request.getEndTime());
+              return new QueryExemplarsFunctionResponseHandle(responseArray);
+            } catch (IOException e) {
+              LOG.error(e.getMessage());
```

Comment:
Just to confirm that logging exception won't leak user sensitive information.

Reply:
No, in my opinion. Do you see any issues? Majority of these exception might result from connection issues/authz/authn issues.